python/pyiceberg/io/pyarrow.py

```diff
@@ -379,3 +385,61 @@ def _(_: StringType) -> pa.DataType:
 def _(_: BinaryType) -> pa.DataType:
     # Variable length by default
     return pa.binary()
+
+
+class _ConvertToArrowExpression(BoundBooleanExpressionVisitor[pc.Expression]):
+    def visit_in(self, term: BoundTerm[pc.Expression], literals: Set[Any]) -> pc.Expression:
+        return pc.field(term.ref().field.name).isin(literals)
+
+    def visit_not_in(self, term: BoundTerm[pc.Expression], literals: Set[pc.Expression]) -> pc.Expression:
+        return ~pc.field(term.ref().field.name).isin(literals)
+
+    def visit_is_nan(self, term: BoundTerm[pc.Expression]) -> pc.Expression:
+        ref = pc.field(term.ref().field.name)
+        return ref.is_null(nan_is_null=True) & ref.is_valid()
+
+    def visit_not_nan(self, term: BoundTerm[pc.Expression]) -> pc.Expression:
+        return ~pc.field(term.ref().field.name).is_null(nan_is_null=True)
```

Comment:
I think that this needs to be the negation of `visit_is_nan` to handle `null` values correctly.

Reply:
Good call!